driver/pelican/types/pelican.go

```diff
@@ -260,51 +300,51 @@ func (pel *Pelican) GetStatus() (*PelicanStatus, error) {
 		return nil, fmt.Errorf("Error retrieving thermostat status from %s: %s", respHist.Request.URL, histResult.Message)
 	}
 
-	if len(histResult.Records.History) == 0 {
-		return nil, nil
-	}
-
-	// Converting string timeStamp to int64 format
-	match := histResult.Records.History[len(histResult.Records.History)-1]
-	timestamp, timeErr := time.ParseInLocation("2006-01-02T15:04", match.TimeStamp, pel.timezone)
-	if timeErr != nil {
-		return nil, fmt.Errorf("Error parsing %v into Time struct: %v\n", match.TimeStamp, timeErr)
-	}
+	if len(histResult.Records.History) > 0 {
```

Comment:
My interpretation is that all this logic is just to check if we've seen thermostat data from the last 2 hours and print a warning if that's true.

So I moved it inside of an `if` to check if the history is non-empty, rather than returning a `nil` status.

Does this seem correct?

Reply:
So this is
- making sure we don't hit an out of bounds error on checking the array
- don't exit the status reading if there's no historical data? We'd return the `PelicanStatus` struct anyway instead of just nil, is how I'm reading this. What's the defined behavior for what's in that struct when we return and the historical data is more than 2 hours old? Is it empty? Old data?

Reply:
The old logic was strange. We did a bunch of work to retrieve the history object, but then all we did was check how old the most recent temperature reading was and, if it was more than 2 hours old, print out a warning message. We did not use this information to modify the `PelicanStatus` struct, i.e. it is decoupled from the thermostat history.

However, if the history object was empty, we returned `nil` for the status. This didn't make sense to me because, even in this situation, we had correctly queried for the thermostat's current status and received a valid result. And, it actually caused a panic in the driver, which is why I made these changes.

Reply:
Ok! Sounds good to me